src/main/java/me/karakelley/http/HttpServer.java

```diff
@@ -37,19 +37,23 @@ public void start() {
     }
   }
 
-  public void startConnection(Socket clientSocket) {
+  private void startConnection(Socket clientSocket) {
     try (LineReader reader = new BufferedLineReader(new InputStreamReader(clientSocket.getInputStream()));
-         PrintWriter out = new PrintWriter(clientSocket.getOutputStream())) {
+         OutputStream out = new DataOutputStream(clientSocket.getOutputStream())) {
       Request newRequest = new Request(reader, clientSocket.getLocalPort());
       sendResponse(newRequest, out);
     } catch (Exception e) {
       logger.info("Ouch!", e);
     }
   }
 
-  public void sendResponse(Request newRequest, PrintWriter out) {
+  private void sendResponse(Request newRequest, OutputStream out) {
     Response response = controller.respond(newRequest);
-    out.write(response.deliver());
+    try {
+      out.write(response.deliver());
+    } catch (IOException e) {
```

Comment:
Same question: describe what would have to happen for an `IOException` to get thrown on the line above.

Reply:
I believe in this case, it would be thrown if the output stream has been closed.